org.lflang/src/org/lflang/generator/c/CGenerator.java

```diff
@@ -1933,6 +1934,18 @@ protected void setUpGeneralParameters() {
             System.out.println("To enable compilation for the Arduino platform, you must specify the fully-qualified board name (FQBN) in the target property. For example, platform: {name: arduino, board: arduino:avr:leonardo}. Entering \"no-compile\" mode and generating target code only.");
             targetConfig.noCompile = true;
         }
+
+        if (targetConfig.platformOptions.platform == Platform.ZEPHYR && targetConfig.platformOptions.userThreads >= 0) {
+            targetConfig.compileDefinitions.put(
+                PlatformOption.USER_THREADS.name(),
+                String.valueOf(targetConfig.platformOptions.userThreads)
+            );
+        }
+
+        if (targetConfig.platformOptions.platform != Platform.ZEPHYR && targetConfig.platformOptions.userThreads >= 0) {
+            System.out.println("Specifying user threads is only for the Zephyr platform. This option will be ignored.");
```

Comment:
Isn`t there an errorReporter object somewhere which can be used to print Warnings.